### PR TITLE
Implemented Check preference of enabling admin advisory banner and retrieval of banner description

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Proposed changes in this pull request
- Change the PreferenceRetrievalClient so that admin-advisory-banner related configs can be retrieved. 
- Two new methods introduced to retreive the two properties and changed
- Created a getConnectorPropertyValue function to retrieve any type of property values.
- Changed the checkPreference method to use the newly created getConnectorPropertyValue method.

### When should this PR be merged
Once the all the related PRs are reviewed and approved, this PR should be able to merge.

### Related Issues
- https://github.com/wso2/product-is/issues/15521
### Related PRs
- https://github.com/wso2-extensions/identity-governance/pull/667
- https://github.com/wso2/identity-apps/pull/3822
- https://github.com/wso2/product-is/pull/15618

### Follow up actions
-

### Checklist (for reviewing)
-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java [hunk 1]
```diff
@@ -238,7 +270,25 @@ public boolean checkPreference(String tenant, String connectorName, String prope
     public boolean checkPreference(String tenant, String connectorName, String propertyName, boolean defaultValue)
             throws PreferenceRetrievalClientException {
 
-        X509HostnameVerifier hv = new SelfRegistrationHostnameVerifier();
+        String propertyValue = getConnectorPropertyValue(tenant, connectorName, propertyName);
+        if (propertyValue != null) {
+            return Boolean.parseBoolean(propertyValue);
+        }
+        return defaultValue;
```
Comment (chamathns) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java:277]: why we are passing a default value as a param? can't we have a static instance for this if it's a default value?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java [hunk 2]
```diff
@@ -264,17 +314,17 @@ public boolean checkPreference(String tenant, String connectorName, String prope
                         JSONObject config = responseProperties.getJSONObject(itemIndex);
                         if (StringUtils.equalsIgnoreCase(
                                 responseProperties.getJSONObject(itemIndex).getString(PROPERTY_NAME), propertyName)) {
-                            return Boolean.valueOf(config.getString(PROPERTY_VALUE));
+                            return config.getString(PROPERTY_VALUE);
                         }
                     }
                 }
-                return defaultValue;
```
Comment (chamathns) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java:271]: where was this previously obtained from?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java [hunk 3]
```diff
@@ -318,7 +368,7 @@ public boolean checkMultiplePreference(String tenant, String connectorName, List
                     JSONArray responseProperties = connector.getJSONArray(PROPERTIES);
                     for (int itemIndex = 0, totalObject = responseProperties.length(); itemIndex < totalObject; itemIndex++) {
                         JSONObject config = responseProperties.getJSONObject(itemIndex);
-                        if (Boolean.valueOf(config.getString(PROPERTY_VALUE))) {
+                        if (Boolean.TRUE.equals(Boolean.valueOf(config.getString(PROPERTY_VALUE)))) {
```
Comment (chamathns) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java:371]:
```suggestion
                        if (Boolean.parseBoolean(config.getString(PROPERTY_VALUE))) {
```

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java (hunk 1 above)
Comment (mpmadhavig) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java:277]: The default value here is a static value (true) and it is common to every property type. 
The value is set from here,

https://github.com/wso2/carbon-identity-framework/blob/12a151508ff2b70a174abc3973f63da851b39985/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java#L257

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java (hunk 2 above)
Comment (mpmadhavig) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java:271]: https://github.com/wso2/carbon-identity-framework/blob/12a151508ff2b70a174abc3973f63da851b39985/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java#L257

**Comment by mpmadhavig** [comment]
The implementation path has been changed and therefore the feature will be sent with the following effort.

1. https://github.com/wso2/carbon-identity-framework/pull/4512
2. https://github.com/wso2/carbon-kernel/pull/3560
3. https://github.com/wso2/identity-api-server/pull/441
4. https://github.com/wso2/identity-apps/pull/3872
5. https://github.com/wso2/product-is/pull/15700

**Pull request closed without merging**